### PR TITLE
Validation file.

### PR DESCRIPTION
**Pull request by jason-villa-accenture**
will be used for validating form.

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
...
+	};
+
+	const isEmpty = (str) => {
+		return str == '';
+	};
+
+	for (const [key, value] of Object.entries(obj)) {
+		if (key.toLowerCase().includes('phone') || key.includes('contact')) {
+			if (!isValidPhone(value)) {
+				result[key] = 'Must be 11-digit that starts with 09';
```
Comment (AccentureJace) [inline src/utils/index.js:35]: Phone Number must be 11 digits and starts with 09

File: src/utils/index.js [hunk 2]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
 		localStorage.removeItem(SESSION_TOKEN);
 	},
 };
+
+export const validateObject = (obj) => {
+	let result = {};
+
+	const isValidPhone = (str) => {
+		const phonePattern = /09\d{9}$/;
```
Comment (AccentureJace) [inline src/utils/index.js:19]: phone_pattern

File: src/utils/index.js [hunk 3]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
...
+		return phonePattern.test(str);
+	};
+
+	const isStrongPassword = (str) => {
+		const passwordPattern = /^(?=(.*[A-Z])(?=.*?[a-z])(?=(.*[0-9]){3,})(?=.*?[#?!@$%^&*-])).{10,}$/;
```
Comment (AccentureJace) [inline src/utils/index.js:24]: password_pattern

File: src/utils/index.js [hunk 4]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
...
+		return passwordPattern.test(str);
+	};
+
+	const isEmpty = (str) => {
+		return str == '';
```
Comment (AccentureJace) [inline src/utils/index.js:29]: specify the variable name. do not use str

File: src/utils/index.js (hunk 1 above)
Comment (AccentureJace) [inline src/utils/index.js:35]: Put all the error messages in constant/common/index.js file

File: src/utils/index.js [hunk 5]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
...
+	for (const [key, value] of Object.entries(obj)) {
+		if (key.toLowerCase().includes('phone') || key.includes('contact')) {
+			if (!isValidPhone(value)) {
+				result[key] = 'Must be 11-digit that starts with 09';
+			}
+		}
+
+		if (key.toLowerCase().includes('password')) {
+			if (!isStrongPassword(value)) {
+				result[key] = 'Password must have atleast 10 characters, 1 special character, 3 numbers and 1 uppercase';
```
Comment (AccentureJace) [inline src/utils/index.js:41]: Put all the error messages in constant/common/index.js file

File: src/utils/index.js [hunk 6]
```diff
@@ -12,3 +11,41 @@ export const JWTStorage = {
...
+			}
+		}
+
+		if (isEmpty(value)) {
+			result[key] = `This field cannot be empty`;
```
Comment (AccentureJace) [inline src/utils/index.js:46]: Put all the error messages in constant/common/index.js file

**Review by AccentureJace — CHANGES_REQUESTED** [review]
@jason-villa-accenture please see request changes

File: src/utils/index.js (hunk 3 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done

File: src/utils/index.js (hunk 1 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done.

File: src/utils/index.js (hunk 5 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done.

File: src/utils/index.js (hunk 6 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done.

File: src/utils/index.js (hunk 4 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done

File: src/utils/index.js (hunk 2 above)
Comment (jason-villa-accenture) [reply src/utils/index.js]: done

File: src/constants/common/index.js [hunk 7]
```diff
@@ -5,3 +5,7 @@ export const BASE_URL = 'https://backend-user-product-management.vercel.app';
 
 export const HEADER_AUTHORIZATION = 'Authorization';
 export const HEADER_GROUP_ID = 'x-request-id';
+
+export const INVALID_PHONE_NUMBER_ERROR = 'Phone Number must be 11-digit that starts with 09';
```
Comment (AccentureJace) [inline src/constants/common/index.js:9]: Phone Number must be 11 digits and starts with 09.

File: src/constants/common/index.js [hunk 8]
```diff
@@ -5,3 +5,7 @@ export const BASE_URL = 'https://backend-user-product-management.vercel.app';
...
+export const INVALID_PASSWORD_ERROR = 'Password must have atleast 10 characters, 1 special character, 3 numbers and 1 uppercase';
+export const EMPTY_FIELD_ERROR = 'This field cannot be empty';
```
Comment (AccentureJace) [inline src/constants/common/index.js:11]: This field is required.

File: src/constants/common/index.js [hunk 9]
```diff
@@ -5,3 +5,7 @@ export const BASE_URL = 'https://backend-user-product-management.vercel.app';
 
 export const HEADER_AUTHORIZATION = 'Authorization';
 export const HEADER_GROUP_ID = 'x-request-id';
+
+export const INVALID_PHONE_NUMBER_ERROR = 'Phone Number must be 11-digit that starts with 09';
+export const INVALID_PASSWORD_ERROR = 'Password must have atleast 10 characters, 1 special character, 3 numbers and 1 uppercase';
```
Comment (AccentureJace) [inline src/constants/common/index.js:10]: Add period after the sentence

**Review by AccentureJace — CHANGES_REQUESTED** [review]
Please see request changes

File: src/constants/common/index.js (hunk 9 above)
Comment (jason-villa-accenture) [reply src/constants/common/index.js:10]: done.

File: src/constants/common/index.js (hunk 7 above)
Comment (jason-villa-accenture) [reply src/constants/common/index.js:9]: done.

**Review by AccentureJace — COMMENTED** [review]
LGTM;

@AccentureAngelo for your review.

Files changes:
- constants/common/index.js
- utils/index.js

**Review by AccentureAngelo — APPROVED** [review]

**Review by AccentureJace — APPROVED** [review]

**Pull request merged**